server/resource_server/service/service.go

```diff
@@ -2,22 +2,32 @@ package service
 
 import (
 	"fmt"
+	"github.com/go-playground/validator/v10"
 	"globe-and-citizen/layer8/server/resource_server/dto"
+	"globe-and-citizen/layer8/server/resource_server/emails/verification"
 	interfaces "globe-and-citizen/layer8/server/resource_server/interfaces"
 	"globe-and-citizen/layer8/server/resource_server/models"
 	"globe-and-citizen/layer8/server/resource_server/utils"
-
-	"github.com/go-playground/validator/v10"
+	"time"
 )
 
 type service struct {
-	repository interfaces.IRepository
+	repository    interfaces.IRepository
+	emailVerifier *verification.EmailVerifier
+
+	verificationCodeValidityDuration time.Duration
 }
 
 // Newservice creates a new instance of service
-func NewService(repo interfaces.IRepository) interfaces.IService {
+func NewService(
+	repo interfaces.IRepository,
+	emailVerifier *verification.EmailVerifier,
```

Comment:
Should this not be an interface? Ah, I see, `EmailVerifier` is a struct but it's `sender.Service` IS an interface. I don't know my Golang well enough to comment on the pattern.

Reply:
Right, for now I decided not to make the EmailVerifier as the interface since we only expect to have a single implementation of email verifications. This might make sense to do it for the sake of simpler unit testing (like to write a mock implementation). In the current flow I tested the real implementation to make extra end-to-end checks.